### DISCUSSION TITLE
Requested Resource NOT Found (livewire/message)

### DISCUSSION THREAD

**Discussion by aichasalhi** [livewire/livewire#4282] [1. Help]
The problem started when I have hosted my web application the server cannot find the requested resource of livewire component and when I click on the button appears not found error
 with this error in the console :index.js:36 POST https://............. 404

II am beginner any help on what's happening will be highly appreciated

**Comment by joshhanley** [comment]
@aichasalhi it looks like your web server isn't configured properly.

The URL should look like `https://mora9yb.com/livewire/message/contry-state-city `, but your URL has `/public` in it.

You shouldn't be pointing your web server to the root directory as that is a security issue, instead it should be pointing directly at your public directory instead.

See Laravel docs here for more details https://laravel.com/docs/8.x/deployment#nginx

> Please ensure, like the configuration below, your web server directs all requests to your application's public/index.php file. You should never attempt to move the index.php file to your project's root, as serving the application from the project root will expose many sensitive configuration files to the public Internet:

Hope this helps!

**Reply by aichasalhi** [reply]
thank you for helping me , the issues resolved by changing `'app_url'` in livewire.php to my web application url .

**Reply by joshhanley** [reply]
@aichasalhi no worries! 

While that fixes the issue, please look at what I said above. Currently the way you app is setup is a serious security risk and so I would strongly recommend changing the way your setup is configured.

But ultimately that's up to you, if security isn't a concern, then changing the app url as you have done is ok 🙂
